apps/web/modules/core/rate-limit/helpers.ts

```diff
@@ -0,0 +1,33 @@
+import { hashString } from "@/lib/hashString";
+import { getClientIpFromHeaders } from "@/lib/utils/client-ip";
+import { logger } from "@formbricks/logger";
+import { checkRateLimit } from "./rate-limit";
+import { type TRateLimitConfig } from "./types/rate-limit";
+
+/**
+ * Get client identifier for rate limiting with IP hashing
+ * Used when the user is not authenticated or the api is called from the client
+ */
+export const getClientIdentifier = async (): Promise<string> => {
```

Comment:
this function throws an error but there'd be no way of knowing this for the consumers of this function, I think we should either move this to a `Result` return type or add JSDoc comments about this throwing an error 🙏